modules/nf-core/mgikit/demultiplex/main.nf

```diff
@@ -0,0 +1,88 @@
+process MGIKIT_DEMULTIPLEX {
+    tag {"$run_id"}
+    label 'process_high'
+
+    conda "${moduleDir}/environment.yml"
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mgikit:0.1.6--h4c94732_0' :
+        'biocontainers/mgikit:0.1.6--h4c94732_0' }"
+
+    input:
+    tuple val(meta), path(samplesheet), path(run_dir)
+
+    output:
+    tuple val(meta), path("${prefix}/*.fastq.gz")                                                    , emit: fastq
+    tuple val(meta), path("${prefix}_undetermined/*.fastq.gz")                                       , optional:true, emit: undetermined
+    tuple val(meta), path("${prefix}_ambiguous/*.fastq.gz")                                          , optional:true, emit: ambiguous
+    tuple val(meta), path("${prefix}/*mgikit.undetermined_barcode*")                                 , emit: undetermined_reports, optional:true
+    tuple val(meta), path("${prefix}/*mgikit.ambiguous_barcode*")                                    , emit: ambiguous_reports, optional:true
+    tuple val(meta), path("${prefix}/*mgikit.general")                                               , emit: general_info_reports
+    tuple val(meta), path("${prefix}/*mgikit.info")                                                  , emit: index_reports
+    tuple val(meta), path("${prefix}/*mgikit.sample_stats")                                          , emit: sample_stat_reports
+    tuple val(meta), path("${prefix}/*mgikit.{info,general,ambiguous_barcode,undetermined_barcode}") , emit: qc_reports
+    path("versions.yml")                                                                             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    run_id = meta.lane ? "${meta.id}-${meta.lane}" : "${meta.id}"
+    prefix = task.ext.prefix ?: "out-${run_id}"
+
+    """
+    out_dir_name="${run_dir}"
+    if [[ "$run_dir" == *.zip ]]; then
```

Comment:
I guess this could be compressed in different manners so I am not sure this should be inside the container

Reply:
This part is only for the testing dataset. Such data in a real scenario is large and the input is a path to a directory.